_includes/ocrtutorial.md

```diff
@@ -0,0 +1,92 @@
+
+The OCR demo has the following code and configuration components, described in more detail later:
+
+-Application logic: The _packages_ directory contains several Nimbella actions, which are logical collections of functions.
+- Static web content: The _web_ directory contains a _public_ directory, with the files that will be part of the public website, and a _src_ directory with React component files and other JavaScript files that are built into the _index.html_ template page to produce the final static public page.
+- Slack application: A Nimbella Commander project folder called _commander_, with an _ocr.js_ file with an app that can be installed into Slack.
+
+- A single index.html file, which has a field for a visitor to enter some text and click **Submit**.
+- A single JavaScript file that provides the backend logic for the conversion of text to QR code.
+- A Node package manager file called *package.json*, which describes what dependencies the function has.
+
+### Project structure and logic
+
+Nimbella relies on directory and file structure to intelligently deploy projects, so the GitHub project directory structure is organized that way and is described below. For more information about creating and deploying Nimbella Cloud projects, see the [Nimbella Command Line Tool (nim) Guide](LINK).
+
+In the following diagrams,  files that contain code or web content are shown with file icons and larger font labels, while project configuration and build files are displayed in smaller fonts without icons.
+
+#### Top-level directories
+
+![](assets/ocrtutorial-44e74a5a.svg)
+
+In any Nimbella project, the deployer looks for one or both of the following top-level directories:
+
+- The _packages_ directory, which contains the back end logic of the project.
+- The _web_ directory, which contains static web content published to be published to the front end of the application.
+
+The _packages_ and _web_ directory contents are described in the following sections.
+
+Any other directories or files that appear at the top level are ignored by the deployer. In this case, there's a third top-level directory called _commander_, which contains logic for a Slack application that is managed by [Nimbella Commander](https://nimbella.com/product/commander).
+
+There are also several configuration files. The _project.yml_ file in the OCR demo configures the object store bucket provided with your Nimbella Cloud namespace for web content and database instances.
+
+#### Actions in the packages directory
+
+Actions are discrete units of functional code in languages such as JavaScript, Node.js, or PHP. Actions are usually divided into packages, and the deployer identifies the packages and actions from the directory and file structure.
+
+In this case, the _packages_ directory contains two  packages with six total actions, shown in this diagram and described below:
+
+![](assets/ocrtutorial-5b2cba72.svg)
+
+  - The `ocr` package contains five actions:
+    - acceptImage
```

Comment:
might be ok to leave the workflow in - it won't work without either a slack token or some commander set up.